tests/cases/fourslash/completionEntryForUnionMethod.ts

```diff
@@ -3,7 +3,7 @@
 ////var y: Array<string>|Array<number>;
 ////y.map/**/(
 
-const text = "(method) Array<T>.map<unknown>(callbackfn: ((value: string, index: number, array: string[]) => unknown) & ((value: number, index: number, array: number[]) => unknown), thisArg?: any): unknown[]";
+const text = "(method) Array<T>.map<unknown>(callbackfn: (value: string | number, index: number, array: (string | number)[]) => unknown, thisArg?: any): unknown[]";

```

Comment:
In the short run user are probably better off being directed to write the single function as shown in the "+" (green),
because it will result in less errors. (See test file `arrayMapPredicates1.ts`, to be uploaded soon).
In the longer tun, we need to talk about overloads and modifying `chooseOverload` to return multiple hits - but that's another story.